node/src/client/mod.rs

```diff
@@ -123,6 +121,10 @@ impl<N: Network, C: ConsensusStorage<N>> Client<N, C> {
             matches!(storage_mode, StorageMode::Development(_)),
         )
         .await?;
+
+        // Initialize the sync module.
+        let sync = BlockSync::new(BlockSyncMode::Router, ledger_service.clone(), router.tcp().clone());
```

Comment:
@raychu86 can you review to confirm this reordering is safe? IIRC this has initialization implications to the node's state.

Reply:
Ditto the above - 

This particular reordering should be safe - `BlockSync::new` now requires an objects from `Router`, which is why it is initialized after Router. `Router` initialization is not affected and does not affect node state.